### PR TITLE
Controller: add account settings page

### PR DESCRIPTION
**Pull request by kevincela**
- Add account settings page for a controller
- Add `useLanguage` composable to handle language switching logic in both controller and standalone mode

### CODE REVIEW

File: public/i18n/en/translation.json [hunk 1]
```diff
@@ -1790,7 +1793,39 @@
...
+      "change_password_description": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Cras consequat, turpis sed facilisis",
+      "ssh_key": "SSH key",
+      "ssh_key_description": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Cras consequat, turpis sed facilisis",
+      "two_factor_authentication": "Two-factor authentication",
+      "two_factor_authentication_description": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Cras consequat, turpis sed facilisis",
+      "two_factor_authentication_disabled": "Two-factor authentication is disabled",
+      "configure_2fa": "Configure 2FA",
+      "generate_ssh_key_pair": "Generate SSH key pair",
+      "new_passphrase": "New passphrase",
+      "new_passphrase_tooltip": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Cras consequat, turpis sed facilisis",
```
Comment (gsanchietti) [inline public/i18n/en/translation.json:1807]:
```suggestion
      "new_passphrase_tooltip": "The passphrase is used to encrypt the private key for enhanced security. It should be a unique mix of letters, numbers, and symbols, hard to guess but easy to remember.",
```

File: public/i18n/en/translation.json [hunk 2]
```diff
@@ -1790,7 +1793,39 @@
       "account_settings_for": "Account settings for '{name}'",
       "change_password": "Change password",
       "language": "Language",
-      "ui_language": "User interface language"
+      "ui_language": "User interface language",
+      "change_language": "Change language",
+      "change_password_description": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Cras consequat, turpis sed facilisis",
+      "ssh_key": "SSH key",
+      "ssh_key_description": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Cras consequat, turpis sed facilisis",
```
Comment (gsanchietti) [inline public/i18n/en/translation.json:1800]:
```suggestion
      "ssh_key_description": "An SSH key pair consists of a public key (shared with servers) and a private key (kept secure locally), enabling secure authentication and data encryption in SSH connections without the need for passwords.",
```

File: public/i18n/en/translation.json [hunk 3]
```diff
@@ -1790,7 +1793,39 @@
       "account_settings_for": "Account settings for '{name}'",
       "change_password": "Change password",
       "language": "Language",
-      "ui_language": "User interface language"
+      "ui_language": "User interface language",
+      "change_language": "Change language",
+      "change_password_description": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Cras consequat, turpis sed facilisis",
```
Comment (gsanchietti) [inline public/i18n/en/translation.json:1798]:
```suggestion
      "change_password_description": "Modify your authentication password for accessing the controller's web interface.",
```

File: public/i18n/en/translation.json [hunk 4]
```diff
@@ -1790,7 +1793,39 @@
...
+      "ssh_key": "SSH key",
+      "ssh_key_description": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Cras consequat, turpis sed facilisis",
+      "two_factor_authentication": "Two-factor authentication",
+      "two_factor_authentication_description": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Cras consequat, turpis sed facilisis",
```
Comment (gsanchietti) [inline public/i18n/en/translation.json:1802]:
```suggestion
      "two_factor_authentication_description": "Two-factor authentication (2FA) adds an extra layer of security by requiring two methods for user verification.",
```

File: public/i18n/en/translation.json [hunk 5]
```diff
@@ -1790,7 +1793,37 @@
       "account_settings_for": "Account settings for '{name}'",
       "change_password": "Change password",
       "language": "Language",
-      "ui_language": "User interface language"
+      "ui_language": "User interface language",
+      "change_language": "Change language",
+      "change_password_description": "Modify your authentication password for accessing the controller's web interface.",
+      "ssh_key": "SSH key",
+      "ssh_key_description": "An SSH key pair consists of a public key (shared with servers) and a private key (kept secure locally), enabling secure authentication and data encryption in SSH connections without the need for passwords.",
```
Comment (andre8244) [inline public/i18n/en/translation.json:1803]:
```suggestion
      "ssh_key_description": "A SSH key pair consists of a public key (shared with servers) and a private key (kept secure locally), enabling secure authentication and data encryption in SSH connections without the need for passwords.",
```

File: public/i18n/en/translation.json [hunk 6]
```diff
@@ -1792,7 +1796,37 @@
...
+      "two_factor_authentication": "Two-factor authentication",
```
Comment (andre8244) [inline public/i18n/en/translation.json:1804]: Can we remove 2FA i18n strings? They are not currently used, and they'll be probably placed inside a distinct i18n section

File: public/i18n/en/translation.json [hunk 7]
```diff
@@ -1792,7 +1796,37 @@
...
+      "minimum_characters": "Minimum 8 characters",
+      "confirm_passphrase": "Confirm passphrase",
+      "generate_key": "Generate key",
+      "ssh_key_generated": "SSH key generated",
+      "ssh_key_removed": "SSH key removed",
+      "passphrase_enabled": "Passphrase enabled",
+      "ssh_public_key": "SSH public key",
+      "remove_key": "Remove key",
+      "remove_key_description": "Remove private key will also remove all public keys for the connected units.",
+      "view_unit_list": "View unit list",
```
Comment (andre8244) [inline public/i18n/en/translation.json:1820]:
```suggestion
      "view_unit_list": "Show unit list",
```

File: src/components/controller/account_settings/ChangePasswordDrawer.vue [hunk 8]
```diff
@@ -0,0 +1,196 @@
...
+function validate() {
+  validationErrorBag.value.clear()
+  let valid = true
+
+  const requiredOldPasswordValidator = validateRequired(oldPassword.value)
+  if (!requiredOldPasswordValidator.valid) {
+    validationErrorBag.value.set('old_password', [
+      requiredOldPasswordValidator.errMessage as string
+    ])
+    valid = false
```
Comment (andre8244) [inline src/components/controller/account_settings/ChangePasswordDrawer.vue:69]: Can we focus the first invalid field?

File: src/components/controller/account_settings/GenerateSSHKeyPairDrawer.vue [hunk 9]
```diff
@@ -0,0 +1,186 @@
...
+}
+
+function validate() {
+  validationErrorBag.value.clear()
+  let valid = true
+
+  const requiredPasswordValidator = validateRequired(passphrase.value)
+  if (!requiredPasswordValidator.valid) {
+    validationErrorBag.value.set('passphrase', [requiredPasswordValidator.errMessage as string])
+    valid = false
```
Comment (andre8244) [inline src/components/controller/account_settings/GenerateSSHKeyPairDrawer.vue:66]: Let's handle focus of invalid fields here too

File: src/components/controller/account_settings/ChangePasswordDrawer.vue [hunk 10]
```diff
@@ -0,0 +1,218 @@
...
+  const requiredPasswordValidator = validateRequired(newPassword.value)
+  if (!requiredPasswordValidator.valid) {
+    validationErrorBag.value.set('new_password', [requiredPasswordValidator.errMessage as string])
+    if (valid) {
+      focusElement(newPasswordRef)
+      valid = false
+    }
+  }
+
+  const requiredConfirmPasswordValidator = validateRequired(newPassword.value)
```
Comment (andre8244) [inline src/components/controller/account_settings/ChangePasswordDrawer.vue:88]:
```suggestion
  const requiredConfirmPasswordValidator = validateRequired(confirmPassword.value)
```

**Review by andre8244 — CHANGES_REQUESTED** [review]

**Review by andre8244 — APPROVED** [review]

**Pull request merged**